x/lockup/keeper/lock.go

```diff
@@ -114,18 +111,36 @@ func (k Keeper) removeTokensFromLock(ctx sdk.Context, lock *types.PeriodLock, co
 	return nil
 }
 
+func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) ([]types.PeriodLock, error) {
+	locks := k.GetAccountLockedDurationNotUnlockingOnly(ctx, owner, coin.Denom, duration)
+	// if existing lock with same duration and denom exists, just add there
+	if len(locks) > 0 {
+		lock := locks[0]
```

Comment:
Is this safe to always add to the first one?

Reply:
I believe so, I didn't change any logic from existing code. And locks usually are in this format: `sdk.Coins{sdk.NewInt64Coin("stake", 5)`